### PR TITLE
Javascript2 week1/catalina

### PR DESCRIPTION
**Pull request by catalina-tech**
# Quiz-App

A simple quiz application built with HTML, CSS, and JavaScript.

- Multiple choice questions
- Score tracking
- Answer shuffling

Check out the live demo of the project: [Quiz App Live Demo](https://catalina-tech.github.io/Quiz-App/)

### CODE REVIEW

File: style.css [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    transition: background-color 0.5s;
+    background: linear-gradient(to right, #845EC2, #FF6F91, #FF9671, #F9F871);
+    background-size: 400% 400%;
+    animation: gradient 15s ease infinite;
+}
+@keyframes gradient {
+    0% { background-position: 0% 50%; }
+    50% { background-position: 100% 50%; }
+    100% { background-position: 0% 50%; }
+}
```
Comment (islanderupnorth) [inline style.css:19]: Nice, looks pretty! Gradients are great 🤩

File: index.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        <div class="quiz">
+            <h2 id="question">What is GMP</h2>
+           <div id="answerButtons">
+            <button class=" btn">Answer 1</button>
+            <button class=" btn">Answer 2</button>
+            <button class=" btn">Answer 3</button>
+            <button class=" btn">Answer 4</button>
+           </div>
+           <button id="btn-next">Next</button>
+           <button id="shuffleButton">shuffle Button</button>
```
Comment (islanderupnorth) [inline index.html:22]: Usually with HTML then `kebab-case` is definitely the most common practice. Having said that I do know a developer or two who prefer Javascript's `camelCase` but the most important thing is to be consistent.

File: script.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    question: "Which of the following is a primary component of Good Manufacturing Practice (GMP) regulations?",
+    answers: [
+            { text: "Maintaining personal hygiene in the workplace", correct: true },
+            { text: "Reducing production costs to increase profit margins", correct: false },
+            { text: "Increasing the number of employees to speed up production", correct: false},
+            { text: "Using the cheapest available raw materials", correct: false },
+    ],
+  },
+  {
+    question: "What is the primary goal of Good Manufacturing Practice (GMP) regulations?",
```
Comment (islanderupnorth) [inline script.js:18]: Missing an `id`

File: script.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+function createQuiz(){
+
+const questions = [
+  { id : 1,
+    question: "Which of the following is a primary component of Good Manufacturing Practice (GMP) regulations?",
+    answers: [
+            { text: "Maintaining personal hygiene in the workplace", correct: true },
+            { text: "Reducing production costs to increase profit margins", correct: false },
+            { text: "Increasing the number of employees to speed up production", correct: false},
+            { text: "Using the cheapest available raw materials", correct: false },
```
Comment (islanderupnorth) [inline script.js:14]: When a key/value pair is a boolean then it is common practice to name it `isCorrect` or `isFalse`. It's not incorrect what you've done but it makes it easier to understand in the code itself for yourself and future collaborators.

File: script.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
+console.log("hello world");
+
+window.addEventListener("DOMContentLoaded", createQuiz);
```
Comment (islanderupnorth) [inline script.js:3]: Nice and short, simply declaring it and passing the function!

File: script.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+}
+
+function showScore(){
+    resetState();
+    questionElement.innerHTML = "Your score is: " + score + " of " + questions.length;
+    nextButton.innerHTML = "Start the quiz again";
+    nextButton.style.display = "block";
+    shuffleButton.style.display = "none";
+
+}
```
Comment (islanderupnorth) [inline script.js:98]: I recommend getting into the habit of using `prettier` and `saveOnFormat`. It helps with all these tiny indentation issues.

File: style.css [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    /* background: #001e4d; */
+    background: #00C9A7;
+    color:#fff;
+    font-weight: 500;
+    border-radius: 10px;
+    border: 0;
+    margin: 20px auto 0;
+    width: 150px;
+    padding:10px;
+    cursor: pointer;
```
Comment (islanderupnorth) [inline style.css:72]: As one of the first things in my CSS files I would actually be default add `cursor: pointer;` to all `button`'s. It more often than not that you want that style on a button.

Right now the question buttons don't have it and I would recommend they do!

**Review by islanderupnorth — COMMENTED** [review]
Really great work Catalina and extra props for your styling - it looks good!

I don't have that much feedback in regards to the Javascript itself as you have applied a lot of the best common practices using the `.classList` to modify elements and and eventListeners. There's no one correct way to do things but your approach is great and very eligible with all the functionality abstracted away into smaller functions.

I would recommend adding the "explanation" functionality to your quiz. Explaining why the answers are correct (plus an extra module to design with CSS!).

Good stuff!

File: index.html (hunk 2 above)
Comment (catalina-tech) [reply index.html:22]: Thank you so much!!!

File: script.js (hunk 4 above)
Comment (catalina-tech) [reply script.js:14]: Great!! I will keep this in mind, thanks

**Review by catalina-tech — COMMENTED** [review]
Thank you so much for your feedback, it means a lot!!

**Pull request merged**